xbmc/epg/EpgInfoTag.cpp

```diff
@@ -488,6 +512,83 @@ std::string CEpgInfoTag::Plot(bool bOverrideParental /* = false */) const
   return retVal;
 }
 
+void CEpgInfoTag::SetOriginalTitle(const std::string &strOriginalTitle)
+{
+  CSingleLock lock(m_critSection);
+  m_bChanged |= (m_strOriginalTitle != strOriginalTitle);
+  m_strOriginalTitle = strOriginalTitle;
+}
+
+std::string CEpgInfoTag::OriginalTitle(bool bOverrideParental /* = false */) const
+{
+  std::string retVal;
+  CSingleLock lock(m_critSection);
+  if (bOverrideParental || !m_pvrChannel || !g_PVRManager.IsParentalLocked(*m_pvrChannel))
+    retVal = m_strOriginalTitle;
+  
+  return retVal;
+}
+
+void CEpgInfoTag::SetCast(const std::string &strCast)
+{
+  CSingleLock lock(m_critSection);
+  m_bChanged |= (m_strCast != strCast);
+  m_strCast = strCast;
+}
+
+std::string CEpgInfoTag::Cast(void) const
+{
+  return m_strCast;
```

Comment:
This is more about pvr potentially loading after the epg db was imported. if we fix the race there, and only import the epg after channels were imported, then we no longer need the setter. It's either set in the ctor then when linked to a channel, or not when the epg is used outside the pvr context, which isn't possible yet